src/vector/transaction.rs

```diff
@@ -177,11 +177,48 @@ impl Dataset {
         }
         Ok(Transaction::new(self))
     }
+
+    /// Optionally start a transaction before running `func` for performance
+    ///
+    /// This uses transaction if the dataset supports it, otherwise it
+    /// runs the `func` function as it is on the dataset. If the
+    /// closure results in error, and it is a transaction, it is
+    /// rolled back. If there is error in rollback it is ignored.
+    pub fn maybe_batch(&mut self, func: impl Fn(&Dataset) -> Result<()>) -> Result<()> {
+        let force = 0; // since this is for speed
+        let rv = unsafe { gdal_sys::GDALDatasetStartTransaction(self.c_dataset(), force) };
+        if rv == OGRErr::OGRERR_UNSUPPORTED_OPERATION {
+            func(self)
+        } else if rv == OGRErr::OGRERR_NONE {
+            let res = func(self);
+            if res.is_ok() {
+                let rv = unsafe { gdal_sys::GDALDatasetCommitTransaction(self.c_dataset()) };
+                if rv != OGRErr::OGRERR_NONE {
+                    Err(GdalError::OgrError {
+                        err: rv,
+                        method_name: "GDALDatasetCommitTransaction",
+                    })
+                } else {
+                    res
+                }
+            } else {
+                let _ = unsafe { gdal_sys::GDALDatasetRollbackTransaction(self.c_dataset()) };
+                // ignoring rollback error
```

Comment:
```suggestion
                // ignore rollback error because it's not guaranteed to be
                // supported, and the `func` failure is more important.
```